recipes/libpng/all/conanfile.py

```diff
@@ -51,31 +66,28 @@ def _patch(self):
                                       'COMMAND "${CMAKE_COMMAND}" -E copy_if_different $<TARGET_LINKER_FILE_NAME:${S_TARGET}> $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/${DEST_FILE}',
                                       'COMMAND "${CMAKE_COMMAND}" -E copy_if_different $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/$<TARGET_LINKER_FILE_NAME:${S_TARGET}> $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/${DEST_FILE}')
 
-        if self.settings.build_type == 'Debug':
-            tools.replace_in_file(os.path.join(self._source_subfolder, 'libpng.pc.in'),
-                                  '-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@',
-                                  '-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@d')
-
     def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["PNG_TESTS"] = "OFF"
-        cmake.definitions["PNG_SHARED"] = self.options.shared
-        cmake.definitions["PNG_STATIC"] = not self.options.shared
-        cmake.definitions["PNG_DEBUG"] = "OFF" if self.settings.build_type == "Release" else "ON"
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["PNG_TESTS"] = "OFF"
+        self._cmake.definitions["PNG_SHARED"] = self.options.shared
+        self._cmake.definitions["PNG_STATIC"] = not self.options.shared
+        self._cmake.definitions["PNG_DEBUG"] = "OFF" if self.settings.build_type == "Release" else "ON"
         if self.settings.os == "Emscripten":
-            cmake.definitions["PNG_BUILD_ZLIB"] = "ON"
-            cmake.definitions["M_LIBRARY"] = ""
-            cmake.definitions["ZLIB_LIBRARY"] = self.deps_cpp_info["zlib"].libs[0]
-            cmake.definitions["ZLIB_INCLUDE_DIR"] = self.deps_cpp_info["zlib"].include_paths[0]
-        if self.settings.os == "Macos":
-            if 'arm' in self.settings.arch:
-                cmake.definitions["PNG_ARM_NEON"] = "on"
+            self._cmake.definitions["PNG_BUILD_ZLIB"] = "ON"
+            self._cmake.definitions["M_LIBRARY"] = ""
+            self._cmake.definitions["ZLIB_LIBRARY"] = self.deps_cpp_info["zlib"].libs[0]
+            self._cmake.definitions["ZLIB_INCLUDE_DIR"] = self.deps_cpp_info["zlib"].include_paths[0]
+        if tools.is_apple_os(self.settings.os):
+            if "arm" in self.settings.arch:
+                self._cmake.definitions["PNG_ARM_NEON"] = "on" if self.settings.os == "Macos" else "off"
```

Comment:
Same remark about using booleans because you're using lowercase values here.

Reply:
https://github.com/glennrp/libpng/blob/a40189cf881e9f0db80511c382292a5604c3c3d1/CMakeLists.txt#L71

Reply:
Consistency is key in libpng's cmake script :rofl: